pkg/eigenState/stakerShares/stakerShares.go

```diff
@@ -610,88 +653,172 @@ func (ss *StakerSharesModel) HandleStateChange(log *storage.TransactionLog) (int
+						and lsd.rn = 1
+				),
+				ranked_staker_shares as (
+					select
+						ds.staker,
+						ss.strategy,
+						COALESCE(ss.shares, 0) as shares,
+						COALESCE(ss.block_number, 0) as block_number, -- Assign a default block number if needed
+						ROW_NUMBER() OVER (
+							PARTITION BY ds.staker, ss.strategy 
+							ORDER BY ss.block_number desc
+						) AS rn
+					from 
+						delegated_stakers as ds
+					left join 
+						staker_shares as ss
+						on ss.staker = ds.staker
+						and ss.strategy = @strategy
+				),
+				latest_staker_shares as (
+					select
+						ls.staker,
+						ls.strategy,
+						ls.shares,
+						ls.block_number
+					from ranked_staker_shares as ls
+					where
+						ls.rn = 1
 				)
-				continue
+				select * from latest_staker_shares
+			`
+			stakerShares := make([]StakerShares, 0)
+
+			// get the staker shares for the stakers who were delegated to the operator
+			// and update the shares with the new max magnitude
+			res := ss.DB.Raw(query,
+				sql.Named("blockNumber", slashDiff.BlockNumber),
+				sql.Named("logIndex", slashDiff.LogIndex),
+				sql.Named("operator", slashDiff.Operator),
+				sql.Named("strategy", slashDiff.Strategy),
+			).Scan(&stakerShares)
+			if res.Error != nil && res.Error != gorm.ErrRecordNotFound {
+				ss.logger.Sugar().Errorw("Failed to fetch staker_shares", zap.Error(res.Error))
+				return nil, res.Error
+			}
+
+			for _, stakerSharesRecord := range stakerShares {
+				slotID := NewShareDeltaSlotID(stakerSharesRecord.Staker, slashDiff.Strategy)
+				var err error
+				currStakerShares, ok := updatedStakerShares[slotID]
+				// if we have not cached but the staker exists in the database for the strategy
+				if !ok && stakerSharesRecord.Strategy == slashDiff.Strategy {
+					currStakerShares, err = stakerSharesRecord.ToStakerSharesDiff()
+					if err != nil {
+						return nil, err
+					}
+				} else if !ok {
+					// otherwise the staker is for a different strategy, so we skip
+					continue
+				}
+
+				diffShares := new(big.Int).Div(new(big.Int).Mul(currStakerShares.Shares, slashDiff.WadsSlashed), WAD)
+				diffShares = diffShares.Mul(diffShares, big.NewInt(-1))
+
+				// append the delta to the delta accumulator
+				ss.deltaAccumulator[blockNumber] = append(ss.deltaAccumulator[blockNumber], &StakerShareDeltas{
+					Staker:          stakerSharesRecord.Staker,
+					Strategy:        slashDiff.Strategy,
+					Shares:          diffShares.String(),
+					StrategyIndex:   0,
+					TransactionHash: slashDiff.TransactionHash,
+					LogIndex:        slashDiff.LogIndex,
+					BlockNumber:     slashDiff.BlockNumber,
+				})
+
+				// update the shares
+				currStakerShares.Shares = currStakerShares.Shares.Add(currStakerShares.Shares, diffShares)
+				currStakerShares.BlockNumber = blockNumber
+
+				// update the local cache
+				updatedStakerShares[slotID] = currStakerShares
 			}
-			prepared.Shares = existingShares.Add(existingShares, newState.Shares)
+		default:
```

Comment:
Can remove this since there is no default action

Reply:
https://github.com/Layr-Labs/sidecar/pull/68/commits/849bd0db77bd553611d6ccdf776de875882dc9c5